x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -342,4 +348,23 @@ export class AlertsClient {
       references,
     };
   }
+
+  private validateActions(alertType: AlertType, actions: Alert['actions']) {
+    // TODO: Should also ensure user has access to each action
+    const { actionGroups: alertTypeActionGroups } = alertType;
+    const usedAlertActionGroups = actions.map(action => action.group);
+    const invalidActionGroups = usedAlertActionGroups.filter(
+      group => !alertTypeActionGroups.includes(group)
+    );
+    if (invalidActionGroups.length) {
+      throw Boom.badRequest(
```

Comment:
I will check with platform team to see what the recommended pattern is for this. I'm sure there's a recommended way to do this with new platform that may not use Boom.

Reply:
Had a chat with platform team regarding this. You are right and Boom usage shouldn't be outside of http endpoint related code. Since we have a few places to change, I have created a card here: https://github.com/elastic/kibana/projects/26#card-26413219 and we can setup the proper functions in the endpoints at the same time.